sdk/core/core-http/lib/util/xml.ts

```diff
@@ -3,24 +3,92 @@
+const xml2jsDefaultOptionsV2 = {
+  explicitCharkey: false,
+  trim: false,
+  normalize: false,
+  normalizeTags: false,
+  attrkey: "$",
+  charkey: "_",
+  explicitArray: true,
+  ignoreAttrs: false,
+  mergeAttrs: false,
+  explicitRoot: true,
+  validator: null,
+  xmlns: false,
+  explicitChildren: false,
+  preserveChildrenOrder: false,
+  childkey: "$$",
+  charsAsChildren: false,
+  includeWhiteChars: false,
+  async: false,
+  strict: true,
+  attrNameProcessors: null,
+  attrValueProcessors: null,
+  tagNameProcessors: null,
+  valueProcessors: null,
+  rootName: "root",
+  xmldec: {
+    version: "1.0",
+    encoding: "UTF-8",
+    standalone: true
+  },
+  doctype: null,
+  renderOpts: {
+    pretty: true,
+    indent: "  ",
+    newline: "\n"
+  },
+  headless: false,
+  chunkSize: 10000,
+  emptyTag: "",
+  cdata: false
+};
+
+// The xml2js settings for general XML parsing operations.
+const xml2jsParserSettings: any = Object.assign({}, xml2jsDefaultOptionsV2);
+xml2jsParserSettings.explicitArray = false;
+xml2jsParserSettings.explicitRoot = false;
+
+// The xml2js settings for ATOM XML parsing operations.
+const xml2jsParserSettingsWithRoot: any = Object.assign({}, xml2jsDefaultOptionsV2);
+xml2jsParserSettingsWithRoot.explicitArray = false;
+
+// The xml2js settings for general XML building operations.
+const xml2jsBuilderSettings: any = Object.assign({}, xml2jsDefaultOptionsV2);
+xml2jsBuilderSettings.explicitArray = false;
+xml2jsBuilderSettings.renderOpts = {
+  pretty: false
+};
+
+/**
+ * Converts given JSON object to XML string
+ * @param obj JSON object to be converted into XML string
+ * @param opts Options that govern the parsing of given JSON object.
+ * `rootName` indicates the name of the root element in the resulting XML
+ */
 export function stringifyXML(obj: any, opts?: { rootName?: string }) {
-  const builder = new xml2js.Builder({
-    explicitArray: false,
-    explicitCharkey: false,
-    rootName: (opts || {}).rootName,
-    renderOpts: {
-      pretty: false
-    }
-  });
-  return builder.buildObject(obj);
+  xml2jsBuilderSettings.rootName = (opts || {}).rootName;
+  const builder = new xml2js.Builder(xml2jsBuilderSettings);
+  return builder.buildObject(removeBOM(obj));
 }
 
-export function parseXML(str: string): Promise<any> {
-  const xmlParser = new xml2js.Parser({
-    explicitArray: false,
-    explicitCharkey: false,
-    explicitRoot: false
-  });
+/**
+ * Converts given XML string into JSON
+ * @param str String containg the XML content to be parsed into JSON
+ * @param opts Options that govern the parsing of given xml string.
+ * `includeRoot` indicates whether the root element is to be included or not in the output
+ */
+export function parseXML(str: string, opts?: { includeRoot?: boolean }): Promise<any> {
+  const xml2jsSettings =
+    opts && opts.includeRoot ? xml2jsParserSettingsWithRoot : xml2jsParserSettings;
```

Comment:
Why have 2 separate constants for parser settings and not do something similar to what we are doing for builder settings? Then, this would simplify to 
```
xml2jsParserSettings.explicitRoot = opts && opts.includeRoot
```

If xml2js needs boolean only and will fail if undefined is passed, then you can use 
```
xml2jsParserSettings.explicitRoot = !!(opts && opts.includeRoot)
```

The !! on undefined will return false

Reply:
Updated